### PR TITLE
Fixes ArchLinux and BlackArch image

### LINKED ISSUE #128: ArchLinux and BlackArch failing
@kiazhi FYI
```
2019-02-08T19:31:21.1715042Z VERBOSE: Running docker  build --pull --quiet --build-arg fromTag=latest --build-arg IMAGE_NAME=pshorg/powershellcommunity:6.1.2-archlinux-latest --build-arg PACKAGE_VERSION=6.1.2 --build-arg PS_VERSION=6.1.2 --build-arg VCS_REF=fe1ad41 -t powershell.local:6.1.2-archlinux-latest -t powershell.local:archlinux-latest /home/vsts/work/1/s/release/community-stable/archlinux/docker
2019-02-08T19:32:19.5916885Z WARNING: *** Last 80 lines of log:
2019-02-08T19:32:19.5917075Z WARNING: Total Download Size:   10.83 MiB
2019-02-08T19:32:19.5917129Z WARNING: Total Installed Size:  42.54 MiB
2019-02-08T19:32:19.5917209Z WARNING: Net Upgrade Size:      41.61 MiB
2019-02-08T19:32:19.5917276Z WARNING: 
2019-02-08T19:32:19.5917320Z WARNING: :: Proceed with installation? [Y/n] 
2019-02-08T19:32:19.5917368Z WARNING: :: Retrieving packages...
2019-02-08T19:32:19.5917948Z WARNING: downloading archlinux-keyring-20190123-1-any.pkg.tar.xz...
2019-02-08T19:32:19.5918219Z WARNING: downloading icu-63.1-2-x86_64.pkg.tar.xz...
2019-02-08T19:32:19.5918467Z WARNING: downloading openssl-1.0-1.0.2.q-1-x86_64.pkg.tar.xz...
2019-02-08T19:32:19.5918558Z WARNING: checking keyring...
2019-02-08T19:32:19.5918605Z WARNING: checking package integrity...
2019-02-08T19:32:19.5918648Z WARNING: loading package files...
2019-02-08T19:32:19.5918692Z WARNING: checking for file conflicts...
2019-02-08T19:32:19.5918772Z WARNING: checking available disk space...
2019-02-08T19:32:19.5918816Z WARNING: :: Processing package changes...
2019-02-08T19:32:19.5919042Z WARNING: reinstalling archlinux-keyring...
2019-02-08T19:32:19.5919136Z WARNING: ==> Appending keys from archlinux.gpg...
2019-02-08T19:32:19.5919185Z WARNING: ==> Locally signing trusted keys in keyring...
2019-02-08T19:32:19.5919451Z WARNING:   -> Locally signing key D8AFDDA07A5B6EDFA7D8CCDAD6D055F927843F1C...
2019-02-08T19:32:19.5919536Z WARNING: ==> ERROR: D8AFDDA07A5B6EDFA7D8CCDAD6D055F927843F1C could not be locally signed.
2019-02-08T19:32:19.5919804Z WARNING:   -> Locally signing key DDB867B92AA789C165EEFA799B729B06A680C281...
2019-02-08T19:32:19.5919866Z WARNING: ==> ERROR: DDB867B92AA789C165EEFA799B729B06A680C281 could not be locally signed.
2019-02-08T19:32:19.5920145Z WARNING:   -> Locally signing key 684148BB25B49E986A4944C55184252D824B18E8...
2019-02-08T19:32:19.5920202Z WARNING: ==> ERROR: 684148BB25B49E986A4944C55184252D824B18E8 could not be locally signed.
2019-02-08T19:32:19.5920455Z WARNING:   -> Locally signing key 91FFE0700E80619CEB73235CA88E23E377514E00...
2019-02-08T19:32:19.5920544Z WARNING: ==> ERROR: 91FFE0700E80619CEB73235CA88E23E377514E00 could not be locally signed.
2019-02-08T19:32:19.5921244Z WARNING:   -> Locally signing key 0E8B644079F599DFC1DDC3973348882F6AC6A4C2...
2019-02-08T19:32:19.5921299Z WARNING: ==> ERROR: 0E8B644079F599DFC1DDC3973348882F6AC6A4C2 could not be locally signed.
2019-02-08T19:32:19.5921700Z WARNING:   -> Locally signing key AB19265E5D7D20687D303246BA1DFB64FFF979E7...
2019-02-08T19:32:19.5921759Z WARNING: ==> ERROR: AB19265E5D7D20687D303246BA1DFB64FFF979E7 could not be locally signed.
2019-02-08T19:32:19.5921839Z WARNING: installing icu...
2019-02-08T19:32:19.5922053Z WARNING: installing openssl-1.0...
2019-02-08T19:32:19.5922265Z WARNING: Optional dependencies for openssl-1.0
2019-02-08T19:32:19.5922467Z WARNING:     ca-certificates [installed]
2019-02-08T19:32:19.5922692Z WARNING: :: Running post-transaction hooks...
2019-02-08T19:32:19.5922740Z WARNING: (1/2) Cleaning up package cache...
2019-02-08T19:32:19.5922783Z WARNING: (2/2) Arming ConditionNeedsUpdate...
2019-02-08T19:32:19.5923008Z WARNING: ‌/bin/sh: tar: command not found‌
2019-02-08T19:32:19.5924474Z WARNING: ‌The command '/bin/sh -c echo "en_US.UTF-8 UTF-8" >> /etc/locale.gen     && locale-gen     && pacman -Syy     && pacman -S --noconfirm       archlinux-keyring       core/icu       openssl-1.0     && mkdir -p ${PS_INSTALL_FOLDER}     && tar zxf /tmp/powershell-linux.tar.gz -C ${PS_INSTALL_FOLDER}     && rm -f /tmp/powershell-linux.tar.gz     && ln -s ${PS_INSTALL_FOLDER}/pwsh /usr/bin/pwsh     && pwsh         -NoLogo         -NoProfile         -Command "           \$ErrorActionPreference = 'Stop' ;           \$ProgressPreference = 'SilentlyContinue' ;           while(!(Test-Path -Path \$env:PSModuleAnalysisCachePath)) {              Write-Host "'Waiting for $env:PSModuleAnalysisCachePath'" ;             Start-Sleep -Seconds 6 ;           }"     && pacman -Syyu --noconfirm     && yes | pacman -Scc' returned a non-zero code: 127‌
2019-02-08T19:32:19.5924645Z WARNING: 
```

### PR DESCRIPTION
**Pull request by kiazhi**
## PR Summary

fixes #128 

- ArchLinux
  - Added `tar` package to uncompress Powershell tar file.
  - Added a fix to `yes | pacman -Scc` command that generates an error because it could not find the `/var/cache/pacman/pkg` directory since it has been automatically removed by the previous `pacman -Syyu --noconfirm` command. The fix is to create that directory before the `yes | pacman -Scc` command is being executed.

- BlackArch
  - Updated the keyring server in the `/tmp/strap.sh` file.
  - Added `tar` package to uncompress Powershell tar file.
  - Added a fix to `yes | pacman -Scc` command that generates an error because it could not find the `/var/cache/pacman/pkg` directory since it has been automatically removed by the previous `pacman -Syyu --noconfirm` command. The fix is to create that directory before the `yes | pacman -Scc` command is being executed.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
  - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `Dockerfile`, `.sh`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
  - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **Adds a new image or tag**
  - [x] Not Applicable
- **OR**
  - [ ] Update [README.powershellcommunity.md](https://github.com/PowerShell/PowerShell-Docker/blob/master/assets/README.powershellcommunity.md)
  - [x] Update [vsts-ci.yml](https://github.com/PowerShell/PowerShell-Docker/blob/master/vsts-ci.yml)

### CODE REVIEW

**Comment by kiazhi** [comment]
Failed the `gss-ntlmssp` pester test and I cannot easily resolve this because `gss-ntlmssp` package is not in the community repo in ArchLinux and therefore it have to be compiled from source using multiple other package dependencies to compile source code using 2 stages dockerfile and copy it.

need a lot code changes in the current dockerfile.

**Comment by RDIL** [comment]
LGTM

**Comment by TravisEz13** [comment]
update the meta.json to skip the gssntlmssp tests

File: release/community-stable/archlinux/docker/Dockerfile [hunk 1]
```diff
@@ -50,6 +50,8 @@ RUN \
       core/icu \
       # required for SSL
       openssl-1.0 \
+      # required for uncompressing tar file
```
Comment (RDIL) [inline release/community-stable/archlinux/docker/Dockerfile]:
```suggestion
      # required for uncompressing tar files
```

File: release/community-stable/blackarch/docker/Dockerfile [hunk 2]
```diff
@@ -63,6 +65,8 @@ RUN \
       core/icu \
       # required for SSL
       openssl-1.0 \
+      # required for uncompressing tar file
```
Comment (RDIL) [inline release/community-stable/blackarch/docker/Dockerfile]:
```suggestion
      # required for uncompressing tar files
```

**Comment by RDIL** [comment]
@TravisEz13 please take a look/merge as this fixes CI

**Comment by TravisEz13** [comment]
@RDIL 	It won't let me commit your suggestion.  I'm merging as is.  Feel free to submit a PR.

**Comment by RDIL** [comment]
Okay 👍

**Pull request merged**